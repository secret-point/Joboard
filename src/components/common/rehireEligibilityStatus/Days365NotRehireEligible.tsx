import React from "react";
import { Col } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import { translate as t } from "../../../utils/translator";
import { Job } from "../../../utils/types/common";

interface Days365NotRehireEligibleProps {
  jobDetail?: Job
}

const Days365NotRehireEligible = (props: Days365NotRehireEligibleProps ) => {

  const { jobDetail } = props;

  const tileWithJobTitle = t("BB-365-days-not-rehire-eligible-title-text-with-jobTitle", "Thank you for your interest in the {jobTitle} position. Our records indicate you previously worked at Amazon and are not eligible for rehire based upon your previous Amazon employment record. As a result, we are unable to move forward with your candidacy at this time.", {jobTitle: jobDetail?.jobTitle || ""});
  const titleWithoutJobTitle = t("BB-365-days-not-rehire-eligible-title-text-without-jobTitle", "Thank you for your interest in this position. Our records indicate you previously worked at Amazon and are not eligible for rehire based upon your previous Amazon employment record. As a result, we are unable to move forward with your candidacy at this time.")

  return (
    <Col gridGap="S300">
      <Text>
        {
          jobDetail ? tileWithJobTitle : titleWithoutJobTitle
        }
      </Text>
      <Text>
        {t("BB-365-days-not-rehire-eligible-content-text", "Due to the termination reason, you can reapply 90 days after your last day worked at Amazon.")}
      </Text>
    </Col>
  )
}

export default Days365NotRehireEligible;
