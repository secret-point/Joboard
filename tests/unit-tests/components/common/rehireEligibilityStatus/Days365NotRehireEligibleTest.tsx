import React from "react";
import { shallow } from "enzyme";
import Days365NotRehireEligible from "../../../../../src/components/common/rehireEligibilityStatus/Days365NotRehireEligible";
import { TEST_JOB2 } from "../../../../test-utils/test-data";


describe("Days365NotRehireEligible", () => {
    it("should match snapshot with jobDetail", () => {
        const shallowWrapper = shallow(<Days365NotRehireEligible jobDetail={ TEST_JOB2 }/>);
    
        expect(shallowWrapper).toMatchSnapshot();
    });

    it("should match snapshot without jobDetail", () => {
        const shallowWrapper = shallow(<Days365NotRehireEligible/>);
    
        expect(shallowWrapper).toMatchSnapshot();
    });
});