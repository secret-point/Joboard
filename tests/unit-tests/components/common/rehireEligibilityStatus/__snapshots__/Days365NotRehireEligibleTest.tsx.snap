// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Days365NotRehireEligible should match snapshot with jobDetail 1`] = `
<Col
  gridGap="S300"
>
  <Text
    font="primary"
    fontSize="T300"
    hyphens="none"
  >
    Thank you for your interest in the {jobTitle} position. Our records indicate you previously worked at Amazon and are not eligible for rehire based upon your previous Amazon employment record. As a result, we are unable to move forward with your candidacy at this time.
  </Text>
  <Text
    font="primary"
    fontSize="T300"
    hyphens="none"
  >
    Due to the termination reason, you can reapply 90 days after your last day worked at Amazon.
  </Text>
</Col>
`;

exports[`Days365NotRehireEligible should match snapshot without jobDetail 1`] = `
<Col
  gridGap="S300"
>
  <Text
    font="primary"
    fontSize="T300"
    hyphens="none"
  >
    Thank you for your interest in this position. Our records indicate you previously worked at Amazon and are not eligible for rehire based upon your previous Amazon employment record. As a result, we are unable to move forward with your candidacy at this time.
  </Text>
  <Text
    font="primary"
    fontSize="T300"
    hyphens="none"
  >
    Due to the termination reason, you can reapply 90 days after your last day worked at Amazon.
  </Text>
</Col>
`;
